examples/cli/src/main/java/io/unitycatalog/cli/UserCli.java

```diff
@@ -0,0 +1,166 @@
+    String externalId = null;
+    if (json.has(CliParams.EXTERNAL_ID.getServerParam())) {
+      externalId = json.getString(CliParams.EXTERNAL_ID.getServerParam());
+    }
+    UserResource userResource =
+        new UserResource()
+            .displayName(json.getString(CliParams.DISPLAY_NAME.getServerParam()))
+            .externalId(externalId)
+            .emails(emails);
+    UserResource user = usersApi.createUser(userResource);
+    return objectWriter.writeValueAsString(User.fromUserResource(user));
+  }
+
+  private static String updateUser(UsersApi usersApi, JSONObject json)
+      throws JsonProcessingException, ApiException {
+    List<Email> emails = new ArrayList<>();
+    String displayName = null;
+    String externalId = null;
+
+    if (json.has(CliParams.EMAIL.getServerParam())) {
+      emails.add(new Email().value(json.getString(CliParams.EMAIL.getServerParam())).primary(true));
+    }
+
+    if (json.has(CliParams.DISPLAY_NAME.getServerParam())) {
+      displayName = json.getString(CliParams.DISPLAY_NAME.getServerParam());
+    }
+
+    if (json.has(CliParams.EXTERNAL_ID.getServerParam())) {
+      externalId = json.getString(CliParams.EXTERNAL_ID.getServerParam());
+    }
+
+    String id = json.getString(CliParams.ID.getServerParam());
+    UserResource userResource =
+        new UserResource().id(id).displayName(displayName).externalId(externalId).emails(emails);
+    UserResource user = usersApi.updateUser(id, userResource);
+    return objectWriter.writeValueAsString(User.fromUserResource(user));
+  }
+
+  private static String listUsers(UsersApi usersApi, JSONObject json)
+      throws JsonProcessingException, ApiException {
+    String filter = null;
+    int startIndex = 1;
+    int count = 100;
+    if (json.has(CliParams.FILTER.getServerParam())) {
+      filter = json.getString(CliParams.FILTER.getServerParam());
+    }
+    if (json.has(CliParams.START_INDEX.getServerParam())) {
+      startIndex = json.getInt(CliParams.START_INDEX.getServerParam());
+    }
+    if (json.has(CliParams.COUNT.getServerParam())) {
+      count = json.getInt(CliParams.COUNT.getServerParam());
+    }
+    List<User> users =
+        usersApi.listUsers(filter, startIndex, count).stream()
+            .map(User::fromUserResource)
+            .collect(Collectors.toList());
+    return objectWriter.writeValueAsString(users);
+  }
+
+  private static String getUser(UsersApi usersApi, JSONObject json)
+      throws JsonProcessingException, ApiException {
+    String id = json.getString(CliParams.ID.getServerParam());
+    UserResource user = usersApi.getUser(id);
+    return objectWriter.writeValueAsString(User.fromUserResource(user));
+  }
+
+  private static String deleteUser(UsersApi usersApi, JSONObject json) throws ApiException {
+    String id = json.getString(CliParams.ID.getServerParam());
+    usersApi.deleteUser(id);
+    return EMPTY;
+  }
+
+  private static class User {
```

Comment:
You _might_ be able to use the User class from the generated models instead of this. I started down this route until I realized that too.